app/src/main/java/com/hamza/Wazaker/Activites/MainActivity/MainActivity.java

```diff
@@ -42,12 +50,19 @@ protected void onCreate(Bundle savedInstanceState) {
 
         binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
 
-        displayNotification();
-
+        fireWorkerTodisplayNotification();
     }
 
-    private void displayNotification() {
-        Notifications.addNotification(this);
+    /**
+     * display notification every 8 hours
+     */
+    private void fireWorkerTodisplayNotification() {
+        WorkRequest WorkRequest =
+                new OneTimeWorkRequest.Builder(NotificationWorker.class)
+                        .setInitialDelay(8, TimeUnit.HOURS)
+                        .build();
+
+        workManager.enqueue(WorkRequest);
```

Comment:
you should use Unique request so not to make new request each time app is opened

Reply:
it's right, Thank you for help and getting the project more cleaner